homeassistant/components/sensor/swiss_hydrological_data.py

```diff
@@ -111,33 +88,35 @@ def __init__(self, hydro_data, measurement, value):
     def name(self):
         """Return the name of the sensor."""
         return "{0} {1} {2}".format(
-            self.hydro_data.data["name"],
-            self._measurement,
+            self._station,
+            self._condition,
             self._value)
 
     @property
     def unit_of_measurement(self):
         """Return the unit of measurement of this entity, if any."""
         if self._state is not STATE_UNKNOWN:
-            return self.hydro_data.data["parameters"][self._measurement]["unit"]
+            return self.hydro_data.data["parameters"][self._condition]["unit"]
         return None
 
     @property
     def state(self):
         """Return the state of the sensor."""
-        try:
+        if isinstance(self._state, (int, float)):
             return round(self._state, 2)
-        except ValueError:
-            return STATE_UNKNOWN
+        return STATE_UNKNOWN
 
     @property
     def state_attributes(self):
         """Return the state attributes."""
+        if not self.hydro_data.data:
+            return {"ATTR_ATTRIBUTION": CONF_ATTRIBUTION}
         return {
             "ATTR_WATER_BODY": self.hydro_data.data["water-body-name"],
             "ATTR_WATER_BODY_TYPE": self.hydro_data.data["water-body-type"],
             "ATTR_STATION": self.hydro_data.data["name"],
-            "ATTR_UPDATE": self.hydro_data.data["parameters"][self._measurement]["datetime"],
+            "ATTR_UPDATE":
+            self.hydro_data.data["parameters"][self._condition]["datetime"],
             "ATTR_ATTRIBUTION": CONF_ATTRIBUTION
         }
```

Comment:
I'm not sure whats the prefered way to set attributes and I'm not very happy with this, especially how they show up in HA (Lovelace UI)
![grafik](https://user-images.githubusercontent.com/948965/48117963-12d30f00-e26b-11e8-8a22-668216cdd256.png)

Reply:
You removed the definition of all constants (line 33-42) and made the key strings in the dict.

Reply:
Ok, got that, I'll fix it today if I find some spare time.